modules/renter/downloadqueue.go

```diff
@@ -60,6 +60,9 @@ func (r *Renter) DownloadQueue() []modules.DownloadInfo {
 			StartTime:   d.startTime,
 		}
 		downloads[i].Received = atomic.LoadUint64(&d.atomicDataReceived)
+		if d.downloadErr != nil {
+			downloads[i].Error = d.downloadErr.Error()
+		}
```

Comment:
This is close but there is one edge case to consider: `renter.Download` can return an error, meaning the download will not be placed in the queue, but `downloadasync` will still return success.